src/signals/settings/users/containers/Detail/hooks/useFetchUser.js

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect } from 'react';
+
+import CONFIGURATION from 'shared/services/configuration/configuration';
+import { getAuthHeaders } from 'shared/services/auth/auth';
+
+export const userEndpoint = `${CONFIGURATION.API_ROOT}signals/v1/private/users/`;
+
+/**
+ * Custom hook useFetchUser
+ *
+ * Will call private /users endpoint
+ *
+ * @returns {FetchResponse}
+ */
+const useFetchUser = id => {
+  const [isLoading, setLoading] = useState();
+  const [data, setData] = useState();
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    const controller = new AbortController();
+    const { signal } = controller;
+
+    async function fetchData() {
+      setLoading(true);
+
+      try {
+        const url = [userEndpoint, id].join('');
+        const response = await fetch(url, {
+          headers: getAuthHeaders(),
+          signal,
+        });
+        const userData = await response.json();
+
+        setData(userData);
+      } catch (e) {
+        setError(e);
+      } finally {
+        setLoading(false);
+      }
+    }
+
+    fetchData();
+
+    return (() => {
+      controller.abort();
```

Comment:
I thought you added an `unmount` test as well... 🤔

Reply:
That was another hook. But well spotted; I'll add that test as well.